src/components/Filters.tsx

```diff
@@ -0,0 +1,126 @@
+import React, { ReactElement, useState, ChangeEvent } from 'react'
+import MUISelect from '@material-ui/core/Select'
+import MenuItem from '@material-ui/core/MenuItem'
+import Box from '@material-ui/core/Box'
+import FormControl from '@material-ui/core/FormControl'
+import InputLabel from '@material-ui/core/InputLabel'
+import uniq from 'lodash/uniq'
+import { withStyles } from '@material-ui/core/styles'
+import { capitalize } from '../helpers'
+import { IProps } from '../containers/FilterContainer'
+import { INamespaceAPI } from '../types/api'
+
+const StyledFormControl = withStyles({
+  root: {
+    margin: '0rem 2rem',
+    minWidth: '10rem'
+  }
+})(FormControl)
+
+type IEntity = INamespaceAPI | string
+interface IFilterDictionary {
+  [key: string]: {
+    entities: IEntity[]
+    accessor: (n: IEntity) => string
+    default: any
+  }
+}
+
+const filterByOptions: { [key: string]: 'namespace' | 'sourceName' } = {
+  /* display name: entitiy name on dataset schema */
+  namespace: 'namespace',
+  datasource: 'sourceName'
+}
+
+const Filters = (props: IProps): ReactElement => {
+  const { namespaces, datasets, filterJobs, filterDatasets, showJobs } = props
+
+  const [currentFilter, setCurrentFilter] = useState('all')
```

Comment:
definitely wouldn't mind a crash course on React hooks (I assume this is what this is?)

Reply:
yes exactly!  Honestly, the docs are pretty great -- https://reactjs.org/docs/hooks-intro.html